src/main/java/de/azapps/kafkabackup/common/partition/PartitionReader.java

```diff
@@ -46,7 +46,9 @@ public void seek(long offset) throws PartitionIndex.IndexException, IOException,
         partitionIndex.seek(offset);
         String segmentFilePrefix = partitionIndex.readFileName();
         currentSegment = new SegmentReader(topic, partition, topicDir, segmentFilePrefix);
-        currentSegment.seek(offset);
+        if (new java.io.File(topicDir.toFile(), segmentFilePrefix).length() != 0) {
```

Comment:
I am thinking whether we should check this here or whether the `SegmentReader` is the correct place. Any ideas from your side?